contributors/design-proposals/daemonset-update.md

```diff
@@ -129,18 +127,35 @@ type DaemonSetStatus struct {
 
 	// UpdatedNumberScheduled is the total number of nodes that are running updated
 	// daemon pod
+	// +optional
 	UpdatedNumberScheduled int32 `json:"updatedNumberScheduled"`
 
 	// NumberAvailable is the number of nodes that should be running the
 	// daemon pod and have one or more of the daemon pod running and
 	// available (ready for at least minReadySeconds)
+	// +optional
 	NumberAvailable int32 `json:"numberAvailable"`
 
 	// NumberUnavailable is the number of nodes that should be running the
 	// daemon pod and have non of the daemon pod running and available
 	// (ready for at least minReadySeconds)
+	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
 }
+
+type DaemonSet struct {
+	// A sequence number representing a specific generation of the template.
+	// Populated by the system. Read-only.
+	// +optional
+	TemplateGeneration int64 `json:"templateGeneration,omitempty"`
+}
+
+const (
+	// DaemonSetTemplateGenerationKey is the key of the labels that is added
+	// to daemon set pods to distinguish between old and new pod templates
+	// during DaemonSet template update.
+	DaemonSetTemplateGenerationKey string = "pod-template-generation"
```

Comment:
See https://github.com/kubernetes/kubernetes/pull/20765#r52097671, 

> Discussed with @bgrant0607 about this, and we renamed it to be without namespace because it's too long whenever we display resources with their labels, and putting namespace is more suitable for annotations because annotations targets machines, not humans.